src/main/java/org/gradle/profiler/mutations/ClearDirectoryMutator.java

```diff
@@ -0,0 +1,67 @@
+package org.gradle.profiler.mutations;
+
+import com.typesafe.config.Config;
+import org.gradle.profiler.BuildMutator;
+import org.gradle.profiler.CompositeBuildMutator;
+import org.gradle.profiler.ConfigUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ClearDirectoryMutator extends AbstractFileSystemMutator {
+
+    private final String description;
+    private final File target;
+    private final List<String> keep;
+
+    public ClearDirectoryMutator(String description, File target, Schedule schedule, List<String> keep) {
+        super(schedule);
+        this.description = description;
+        this.target = target;
+        this.keep = keep;
+    }
+
+    @Override
+    protected void executeOnSchedule() {
+        System.out.printf("> Clearing %s: '%s'%n", description, target.getAbsolutePath());
+        if (!target.exists()) {
+            return;
+        }
+
+        if (!target.isDirectory()) {
+            throw new IllegalArgumentException(String.format("Cannot clear '%s' since it is not a directory", target.getAbsolutePath()));
+        }
+
+        try (Stream<Path> contents = Files.list(target.toPath())) {
+            contents
+                .filter(path -> !keep.contains(path.getFileName().toString()))
```

Comment:
❔  Is my understanding correct we currently keep only top level files/directories and not files or directories on lower level. I think it's fine for now, just would like to understand it

Reply:
See comment above